### PR TITLE
fix(data-table): migration to tegel

### PR DESCRIPTION
**Pull request by theJohnnyMe**
**Describe pull-request**  
Migration of data-table component

**Solving issue**  
Fixes: [AB#2455](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2455)

**How to test**  
1. Open the link from Netlify Storybook
2. Click around data-table and see if all stories render properly. There should be 10 stories in total including the native one.

**Additional context**  
Be aware that the data-table consists of 9 subcomponents so theming is not done in this PR as it involves a lot of work. In order to proceed with work on the data-table, just a basic migration of controls and readme files was done. other segments of migration will be done in separate PRs.

### CODE REVIEW

**Comment by mJarsater** [comment]
I am getting some curly crackets in the code example that I don't think are suppose to be there.

Screenshot 2022-11-10 at 10 32 18

File: tegel/src/components/data-table/table-component-basic.stories.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      vertical-dividers="${args.verticalDivider}"
+      compact-design="${args.compactDesign}"
+      white-background="${args.onWhiteBackground}"
+      enable-responsive="${args.responsiveDesign}"
+            >
+      <sdds-table-header>
+          <sdds-header-cell column-key='truck' column-title='Truck type' }></sdds-header-cell>
+          <sdds-header-cell column-key='driver' column-title='Driver name' }></sdds-header-cell>
+          <sdds-header-cell column-key='country' column-title='Country' }></sdds-header-cell>
+          <sdds-header-cell column-key='mileage' column-title='Mileage' text-align='right' }></sdds-header-cell>
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-component-basic.stories.ts:92]: Remove curly brackets.

File: tegel/src/components/data-table/table-component-basic.stories.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    notes: {
+      'sdds-table': sddsTable,
+      'sdds-table-toolbar': sddsTableToolbar,
+      'sdds-header': sddsHeader,
+      'sdds-header-cell': sddsHeaderCell,
+      'sdds-table-body': sddsTableBody,
+      'sdds-body-row': sddsBodyRow,
+      'sdds-body-row-expandable': sddsBodyRowExpandable,
+      'sdds-body-cell': sddsBodyCell,
+      'sdds-table-footer': sddsTableFooter,
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-component-basic.stories.ts:24]: Really nice way of handling multiple readme's!

File: tegel/src/components/data-table/table-component-event-listeners.stories.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    document.getElementById('event-name-textarea').value = 'sortColumnDataEvent';
+    document.getElementById('event-value-textarea').value = e.detail;
+  });
+
+  window.addEventListener('currentPageValueEvent', e => {
+    document.getElementById('event-name-textarea').value = 'currentPageValueEvent';
+    document.getElementById('event-value-textarea').value = e.detail;
+  });
+</script>
+  <h3>Disabled filtering, pagination and sorting - left to the user to listen to events</h3>
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-component-event-listeners.stories.ts:98]: Maybe add a subheadline here informing the user this only work  in the canvas tab?

File: tegel/src/components/data-table/table-component-multiselect.stories.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    });
+  }
+  window.addEventListener('click', () => {
+    getValue();
+  });
+
+</script>
+
+  <h3>Multiselect</h3>
+   <sdds-table
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-component-multiselect.stories.ts:109]: Maybe add a subheadline here informing the user this only work in the canvas tab?

File: tegel/src/components/data-table/table-component-event-listeners.stories.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                  <sdds-body-cell cell-value="Test value 6" cell-key="driver"></sdds-body-cell>
+                  <sdds-body-cell cell-value="Test value 7" cell-key="country"></sdds-body-cell>
+                  <sdds-body-cell cell-value="Test value 8" cell-key="mileage"></sdds-body-cell>
+              </sdds-table-body-row>
+          </sdds-table-body>
+        <sdds-table-footer enable-client-pagination client-max-pages="10"></sdds-table-footer>
+  </sdds-table>
+  <br>
+  <div style="width: 500px; background-color: lightblue; padding: 16px;">
+    <h5>Event test box</h5>
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-component-event-listeners.stories.ts:125]: Maybe add a subheadline here informing the user this only work in the canvas tab?

**Review by mJarsater — CHANGES_REQUESTED** [review]
Remove unnecessary curly brackets on the <sdds-header-cell> on multiple stories. Otherwise it looks good to me, really nice work!

File: tegel/src/components/data-table/table-component-basic.stories.ts (hunk 2 above)
Comment (theJohnnyMe) [reply tegel/src/components/data-table/table-component-basic.stories.ts:24]: Yeah, Emelie did it for Dropdown first as it is also more than one component and I felt it makes sense to have all of them in all stories so users have access to it.

File: tegel/src/components/data-table/table-component-multiselect.stories.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+              <sdds-header-cell column-key='country' column-title='Country'></sdds-header-cell>
+              <sdds-header-cell column-key='mileage' column-title='Mileage' text-align='right'></sdds-header-cell>
+          </sdds-table-header>
+          <sdds-table-body enable-dummy-data>
+          </sdds-table-body>
+  </sdds-table>
+  <br>
+  <div style="width: 500px; background-color: lightblue; padding: 16px;">
+
+    <h6 class="sdds-u-pb0 sdds-u-mb0">Selected rows data</h6>
```
[the remark this answers is not in the document]
Comment (theJohnnyMe) [reply tegel/src/components/data-table/table-component-multiselect.stories.ts:129]: Absolutely, good call!

File: tegel/src/components/data-table/table-component-event-listeners.stories.ts (hunk 5 above)
Comment (theJohnnyMe) [reply tegel/src/components/data-table/table-component-event-listeners.stories.ts:125]: Absolutely, good call!

**Comment by theJohnnyMe** [comment]
@mJarsater  - Thanks for review buddy, I think all comments are sorted out now, please double check ;)

**Review by mJarsater — APPROVED** [review]
Nice!

**Pull request merged**